### PR TITLE
Introduce docker standalone documentation, refactor intro

### PR DESCRIPTION
**Pull request by actionbuddha**
This PR introduces some installation documentation for StorageOS within Docker standalone (while emphasising that it is for test purposes only). The value of this is that it lowers the bar for new users of StorageOS, many of whom won't have a handy Openshift/Kubernetes cluster to test with, but might like the notion of persistent storage for containers. As a new joiner, this was the first thing I tried to do (independent of any test VMs, which aren't likely to get used in any real capacity).

Some other tidy ups - the introduction section gets a re-organisation, broken down into overview/quickstart/support with an explanation of our main doc headings, and the environment variables section is moved to Reference, where it belongs.

### CODE REVIEW

File: _docs/platforms/docker/install.md [hunk 1]
```diff
@@ -7,10 +7,49 @@ redirect_from: /docs/install/docker/container
...
+    --env=ADVERTISE_IP=127.0.0.1 \
+    --env=JOIN=127.0.0.1 \
+    --net=host \
+    --pid=host \
+    --privileged \
+    --cap-add=SYS_ADMIN \
+    --device=/dev/fuse \
+    --volume=/var/lib/storageos:/var/lib/storageos:rshared \
+    --volume=/run/docker/plugins:/run/docker/plugins \
+    --volume=/sys:/sys
```
Comment (Arau) [inline _docs/platforms/docker/install.md]: There is one \ missing at the end of this line

Comment (actionbuddha) [reply _docs/platforms/docker/install.md]: Nice catch - thanks!

File: _docs/introduction/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+---
+
+# Quickstart
+
+* Start by [understanding the architecture]({% link
+_docs/concepts/architecture.md %}) and [playing with a demo](http://play.storageos.com/main).
+
+* For test deployments to gain a quick overview of the product, we offer a
+   simple method to run Storage OS with [Docker]({% link
+_docs/platforms/docker/install.md %}.)
```
Comment (croomes) [inline _docs/introduction/quickstart.md]: broken link: move '.' after closing ')'

File: _docs/introduction/support.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+---
+
+# Support
+
+There are several ways to reach us if you require support. The fastest way to
+get in touch is to [join our public Slack
+channel.](https://slack.storageos.com) <script async defer
+src="http://slack.storageos.com/slackin.js"></script>
+
+You can file a support ticket by [emailing
```
Comment (croomes) [inline _docs/introduction/support.md]: I'd remove "emailing" from the link

File: _docs/platforms/docker/install.md [hunk 4]
```diff
@@ -7,10 +7,49 @@ redirect_from: /docs/install/docker/container
...
+    --env=JOIN=127.0.0.1 \
+    --net=host \
+    --pid=host \
+    --privileged \
+    --cap-add=SYS_ADMIN \
+    --device=/dev/fuse \
+    --volume=/var/lib/storageos:/var/lib/storageos:rshared \
+    --volume=/run/docker/plugins:/run/docker/plugins \
+    --volume=/sys:/sys \
+    storageos/node:1.0.0-rc4 server
```
Comment (croomes) [inline _docs/platforms/docker/install.md]: Use:
```
  storageos/node:{{ site.latest_node_version }} server
```
See https://github.com/storageos/storageos.github.io/pull/193/commits/f70cef581bf0c0536ff873747d30602fe8f57a0a (the PR got closed, but I don't think it was applied everywhere?)

File: _docs/platforms/docker/install.md [hunk 5]
```diff
@@ -7,10 +7,49 @@ redirect_from: /docs/install/docker/container
...
-If you are running containers without orchestration, you can create a systemd
-service to run the StorageOS container on each node.
+* Ensure that our prerequisites are met, paying particular attention to [system
+  configuration]({% link _docs/prerequisites/systemconfiguration.md %}) and [mount
+  propagation]({% link _docs/prerequisites/mountpropagation.md %})
+* Run a docker container as follows:
+```bash
+/usr/bin/docker run \
+    --name=storageos \
+    --env=DISABLE_TELEMETRY=true \
```
Comment (croomes) [inline _docs/platforms/docker/install.md]: Remove `DISABLE_TELEMETRY`?

File: _docs/platforms/docker/install.md [hunk 6]
```diff
@@ -22,4 +61,3 @@ nano hosts
 ansible-playbook -i hosts site.yaml
 ```
 
-{% include envvars.md %}
```
Comment (croomes) [inline _docs/platforms/docker/install.md]: My preference is that wherever we're showing commands that use envvars, we provide an explanation of what they are.  The include seemed like a good way of doing this - not sure the motivation behind removing it.

Comment (actionbuddha) [reply _docs/platforms/docker/install.md]: The PR moves the environment variable stuff to the reference section as a first class citizen. It feels neater that way and (IMHO) more intuitive. I can add a note to the various install sections to explain it but it feels superfluous...

**Pull request merged**